samples/nrf9160/location/src/main.c

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <string.h>
+#include <zephyr.h>
+#include <nrf_modem_at.h>
+#include <modem/lte_lc.h>
+#include <modem/location.h>
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+#include <date_time.h>
+#endif /* CONFIG_NRF_CLOUD_PGPS */
+
+static struct loc_config config;
+static struct loc_method_config methods[2];
+
+K_SEM_DEFINE(location_event, 0, 1);
+
+static void antenna_configure(void)
+{
+	int err;
+
+	if (strlen(CONFIG_LOCATION_SAMPLE_GNSS_AT_MAGPIO) > 0) {
+		err = nrf_modem_at_printf("%s", CONFIG_LOCATION_SAMPLE_GNSS_AT_MAGPIO);
+		if (err) {
+			printk("Failed to set MAGPIO configuration\n");
+		}
+	}
+
+	if (strlen(CONFIG_LOCATION_SAMPLE_GNSS_AT_COEX0) > 0) {
+		err = nrf_modem_at_printf("%s", CONFIG_LOCATION_SAMPLE_GNSS_AT_COEX0);
+		if (err) {
+			printk("Failed to set COEX0 configuration\n");
+		}
+	}
+}
+
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+K_SEM_DEFINE(time_update_finished, 0, 1);
+
+static void date_time_evt_handler(const struct date_time_evt *evt)
+{
+	/* Don't care about the event, just indicate that we got it. */
+	k_sem_give(&time_update_finished);
+}
+
+static void current_time_get(void)
+{
+	int err;
+
+	/* If we wait for a while after LTE attach, it's more likely that modem has got the current
+	 * time from the LTE network and no NTP is needed.
+	 */
+	k_sleep(K_SECONDS(1));
+
+	err = date_time_update_async(date_time_evt_handler);
```

Comment:
Haven't checked what the library does for P-GPS but I'm thinking why library is not handling this date_time issue but application using P-GPS has to worry about it?

Reply:
I think this is the right place for updating the time. 😄 And libraries shouldn't be calling `date_time_update_async()` by themselves. That may cause data traffic (when LTE network doesn't provide the time), so the application should be in control.

Reply:
As discussed, I still think application shouldn't care about the time updates.
As we discussed, we'll try to implement this stuff into date_time lib so app shouldn't need to worry about it.